cli/main.ts

```diff
@@ -0,0 +1,86 @@
+import { Command } from "https://deno.land/x/cliffy@v0.25.4/command/mod.ts";
+import {
+  DenoLandProvider,
+  UpgradeCommand,
+} from "https://deno.land/x/cliffy@v0.25.4/command/upgrade/mod.ts";
+import login from "./login.ts";
+import flow from "./flow.ts";
+import script from "./script.ts";
+import workspace from "./workspace.ts";
+import resource from "./resource.ts";
+import remote from "./remote.ts";
+import user from "./user.ts";
+import setup from "./setup.ts";
+import variable from "./variable.ts";
+import push from "./push.ts";
+
+await new Command()
+  .name("windmill")
+  .description("A simple CLI tool for windmill.")
+  .globalOption(
+    "--base-url <baseUrl:string>",
+    "Specify the base url to use when interacting with the API.",
+    {
+      conflicts: ["remote"],
+    }
+  )
+  .globalOption(
+    "--workspace <workspace_id:string>",
+    "Specify the target workspace. This overrides the default workspace."
+  )
+  .globalOption(
+    "--remote <remote_name:string>",
+    "Specify the target remote, add to this list via `wmill remote add`.",
+    {
+      conflicts: ["base-url"],
+    }
+  )
+  .globalOption(
+    "--token <token:string>",
+    "Specify a token to use for authentication. This will not be stored. Takes presedence over username/password",
+    {
+      conflicts: ["email", "password"],
+    }
+  )
+  .globalOption(
+    "--email <email:string>",
+    "Specify credentials to use for authentication. This will not be stored. It will only be used to exchange for a token with the API server, which will not be stored either.",
+    {
+      depends: ["password"],
+      conflicts: ["token"],
+    }
+  )
+  .globalOption(
+    "--password <password:string>",
+    "Specify credentials to use for authentication. This will not be stored. It will only be used to exchange for a token with the API server, which will not be stored either.",
+    {
+      depends: ["email"],
+      conflicts: ["token"],
+    }
+  )
+  .version("v0.0.0")
```

Comment:
we might want to not include version or if possible use the one from version.txt at the root

Reply:
I don't think deno has a way to read the version "compile" time.
We'd have to set it manually.
Could also just put 1.0.0 and then version the CLI separately.